presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/BingTileFunctions.java

```diff
@@ -190,6 +193,136 @@ public static Block bingTilesAround(
         return blockBuilder.build();
     }
 
+    @Description("Given a (longitude, latitude) point, returns the surrounding Bing tiles " +
+            "at the specified zoom level covering minimum radiusInKm")
+    @ScalarFunction("bing_tiles_around")
+    @SqlType("array(" + BingTileType.NAME + ")")
+    public static Block bingTilesAround(
+            @SqlType(StandardTypes.DOUBLE) double latitude,
+            @SqlType(StandardTypes.DOUBLE) double longitude,
+            @SqlType(StandardTypes.INTEGER) long zoomLevel,
+            @SqlType(StandardTypes.DOUBLE) double radiusInKm)
+    {
+        checkLatitude(latitude, LATITUDE_OUT_OF_RANGE);
+        checkLongitude(longitude, LONGITUDE_OUT_OF_RANGE);
+        checkZoomLevel(zoomLevel);
+
+        long mapSize = mapSize(toIntExact(zoomLevel));
+        long maxTileIndex = (mapSize / TILE_PIXELS) - 1;
+
+        long tileY = longitudeToTileY(latitude, mapSize);
+        long tileX = longitudeToTileX(longitude, mapSize);
+
+        // Find top, bottom, left and right tiles from center of circle
+        double[] topLatLong = GeoFunctions.addDistanceToLatitudeLongitude(latitude, longitude, radiusInKm, 0);
+        BingTile topTile = latitudeLongitudeToTile(correctLatitude(topLatLong[0]),
+                correctLongitude(topLatLong[1]),
+                toIntExact(zoomLevel));
+
+        double[] bottomLatLong = GeoFunctions.addDistanceToLatitudeLongitude(latitude, longitude, radiusInKm, 180);
+        BingTile bottomTile = latitudeLongitudeToTile(correctLatitude(bottomLatLong[0]),
+                correctLongitude(bottomLatLong[1]),
+                toIntExact(zoomLevel));
+
+        double[] leftLatLong = GeoFunctions.addDistanceToLatitudeLongitude(latitude, longitude, radiusInKm, 270);
+        BingTile leftTile = latitudeLongitudeToTile(correctLatitude(leftLatLong[0]),
+                correctLongitude(leftLatLong[1]),
+                toIntExact(zoomLevel));
+
+        double[] rightLatLong = GeoFunctions.addDistanceToLatitudeLongitude(latitude, longitude, radiusInKm, 90);
+        BingTile rightTile = latitudeLongitudeToTile(correctLatitude(rightLatLong[0]),
+                correctLongitude(rightLatLong[1]),
+                toIntExact(zoomLevel));
+
+
+        Set<BingTile> excludeTiles = new HashSet<>();
```

Comment:
I feel that building a set of tiles to exclude and then looking up every tile adds unnecessary CPU cost. I added a simple benchmark to see if that's the case. 

[BenchmarkBingTilesAround.java.txt](https://github.com/prestodb/presto/files/2255935/BenchmarkBingTilesAround.java.txt)

The benchmark is calling `bing_tile_around` function for zoom levels 10, 14 and 20 and radiuses 0.1, 1 and 5. It should that the function becomes significantly slower at zoom level 20.

```
Benchmark                           (radiusKm)  (zoomLevel)  Mode  Cnt   Score    Error  Units
BenchmarkBingTilesAround.benchmark         0.1           10  avgt   10   0.002 ±  0.001  ms/op
BenchmarkBingTilesAround.benchmark         0.1           14  avgt   10   0.002 ±  0.001  ms/op
BenchmarkBingTilesAround.benchmark         0.1           20  avgt   10   0.008 ±  0.001  ms/op
BenchmarkBingTilesAround.benchmark           1           10  avgt   10   0.002 ±  0.001  ms/op
BenchmarkBingTilesAround.benchmark           1           14  avgt   10   0.002 ±  0.001  ms/op
BenchmarkBingTilesAround.benchmark           1           20  avgt   10   0.402 ±  0.027  ms/op
BenchmarkBingTilesAround.benchmark           5           10  avgt   10   0.002 ±  0.001  ms/op
BenchmarkBingTilesAround.benchmark           5           14  avgt   10   0.011 ±  0.001  ms/op
BenchmarkBingTilesAround.benchmark           5           20  avgt   10  11.375 ±  2.881  ms/op
```

I then updated the logic to remove `excludeTiles` set and saw the slowest benchmark improve 2x.

[BingTileFunctions.java.txt](https://github.com/prestodb/presto/files/2255942/BingTileFunctions.java.txt)


```
Benchmark                           (radiusKm)  (zoomLevel)  Mode  Cnt  Score    Error  Units
BenchmarkBingTilesAround.benchmark         0.1           10  avgt   10  0.002 ±  0.001  ms/op
BenchmarkBingTilesAround.benchmark         0.1           14  avgt   10  0.002 ±  0.001  ms/op
BenchmarkBingTilesAround.benchmark         0.1           20  avgt   10  0.007 ±  0.001  ms/op
BenchmarkBingTilesAround.benchmark           1           10  avgt   10  0.002 ±  0.001  ms/op
BenchmarkBingTilesAround.benchmark           1           14  avgt   10  0.002 ±  0.001  ms/op
BenchmarkBingTilesAround.benchmark           1           20  avgt   10  0.339 ±  0.007  ms/op
BenchmarkBingTilesAround.benchmark           5           10  avgt   10  0.002 ±  0.001  ms/op
BenchmarkBingTilesAround.benchmark           5           14  avgt   10  0.011 ±  0.001  ms/op
BenchmarkBingTilesAround.benchmark           5           20  avgt   10  6.739 ±  0.238  ms/op
```

Reply:
whats the suggestions here?
1. To add exclude logic during the building of all tiles? I believe that will be slower but we can test for it.
2. Not to exclude tiles?

Reply:
@avinashidnani Currently the code iterates over all tiles and collects tiles to exclude into a set, then iterates again, checks tiles against tiles-to-exclude set and builds the result. I suggest to combine the two passes into one and eliminate the tiles-to-exclude set. I attached the modified code and a benchmark that shows performance gains.

Reply:
Got it, I'll make that change. Thanks!